test/e2e/networkpolicy_test.go

```diff
@@ -91,6 +91,10 @@ func TestNetworkPolicy(t *testing.T) {
 		t.Cleanup(exportLogsForSubtest(t, data))
 		testIngressPolicyWithEndPort(t, data)
 	})
+	t.Run("testNamespaceIsolation", func(t *testing.T) {
+		t.Cleanup(exportLogsForSubtest(t, data))
+		testNamespaceIsolation(t, data)
+	})
```

Comment:
Haven't namespace isolation already covered by other tests? I feel the new test should focus on haripin case more, to avoid duplication and explicitly show what it is for. Otherwise it could be removed by accident when someone refactors the test.
It could just create two Pods and a Service backed by one of them, apply a deny-all policy, validate only the Pod itself can access it and the other one can't.

Reply:
Changed. But I used a deny-ingress policy, instead of a deny-all policy, since the egress deny rule still can drop this traffic.
Also, I changed a little bit on the `toService` test case covering egress rule enforced on hairpin traffic.